### PR TITLE
[ELY-1843] - Adding ability to specify custom ScheduledExecutorService and to shut down ScheduledExecutorService in NonceManager

### PR DESCRIPTION
**Pull request by ashley-abdelsayed98**
https://issues.jboss.org/browse/ELY-1843

### CODE REVIEW

File: http/base/src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java [hunk 1]
```diff
@@ -49,4 +49,7 @@
      */
     HttpServerAuthenticationMechanism createAuthenticationMechanism(String mechanismName, Map<String, ?> properties, CallbackHandler callbackHandler) throws HttpAuthenticationException;
 
+    default void shutdown() {
```
Comment (fjuma) [inline http/base/src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: Would be good to get some javadoc added here.

File: http/digest/src/main/java/org/wildfly/security/http/digest/NonceManager.java [hunk 2]
```diff
@@ -271,6 +273,13 @@ boolean useNonce(final String nonce, byte[] salt, int nonceCount) throws Authent
         }
     }
 
+    /**
+     * must be called in order to shut down executor
+     */
+    public void shutDown() {
```
Comment (fjuma) [inline http/digest/src/main/java/org/wildfly/security/http/digest/NonceManager.java]: This is a very minor comment, but noticed that "shutDown" is used here but "shutdown" is used elsewhere. Would be good to be consistent.

Comment (ashley-abdelsayed98) [reply http/digest/src/main/java/org/wildfly/security/http/digest/NonceManager.java]: Fixed

File: http/base/src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java (hunk 1 above)
Comment (ashley-abdelsayed98) [reply http/base/src/main/java/org/wildfly/security/http/HttpServerAuthenticationMechanismFactory.java]: Fixed

**Pull request merged**